src/game/server/swarm/asw_marine.cpp

```diff
@@ -4251,6 +4261,31 @@ void CASW_Marine::Event_Killed( const CTakeDamageInfo &info )
 	}
 
 	m_bSlowHeal = false;	// no healing if we're dead!
+
+	CASW_Inhabitable_NPC* pAttackerNPC = dynamic_cast< CASW_Inhabitable_NPC* >( pAttacker );
+	if ( pAttackerNPC && pAttackerNPC->Classify() != CLASS_ASW_MARINE )
+	{
+		if ( m_bSpawnZombineOnDeath || pAttackerNPC->m_bSpawnZombineOnMarineKill || rd_marine_spawn_zombine_on_death_chance.GetFloat() > RandomFloat() )
+		{
+			CNPC_Zombine* pZombine = dynamic_cast< CNPC_Zombine* >( CreateEntityByName( "npc_zombine" ) );
+
+			if ( !pZombine )
+			{
+				Warning( "Unable to create npc_zombine from dying marine." );
+			}
+			else
+			{
+				pZombine->SetAbsOrigin( GetAbsOrigin() + Vector( 0.0, 0.0, 4.0 ) );	// spawns inside the ground, add 4 to height
+
+				pZombine->Spawn();
+
+				pZombine->SetModel( "models/zombie/fallen_marine.mdl" );
+				pZombine->m_nSkin = GetSkin();
+				// todo: color the zombine in same color
+				//pZombine->SetRenderColor( GetRenderColor() );	
```

Comment:
You probably also want to set the body group to either have the purple glob on their back visible or turn their original backpack back on.